<launch>

    <arg name="first_robot"     default="tb3_0"/>
    <arg name="second_robot"    default="tb3_1"/>
    <arg name="third_robot"     default="tb3_2"/>
        
    <!-- first robot -->
    <group ns="$(arg first_robot)">        
        
        <!-- Bringup -->
        <include file="$(find turtlebot3_bringup)/launch/turtlebot3_remote.launch">
            <arg name="multi_robot_name" value="$(arg first_robot)"/>
        </include>

        <!-- Map server -->
        <node name="map_server" pkg="map_server" type="map_server" args="$(find robot_navigation)/maps/office.yaml" />

        <!-- AMCL -->
        <include file="$(find robot_navigation)/launch/amcl_custom.launch">
            <arg name="initial_pose_x"  value="-7.4"/>
            <arg name="initial_pose_y"  value="-0.9"/>
            <arg name="initial_pose_a"  value="1.57"/>
            <arg name="ROS_NAMESPACE"   value="$(arg first_robot)" />
        </include>

        <!-- Move Base -->
        <include file="$(find robot_navigation)/launch/move_base_custom.launch">
            <arg name="cmd_vel_topic" value="cmd_vel"/>
            <arg name="ROS_NAMESPACE" value="$(arg first_robot)" />
        </include>

    </group>


    <!-- second robot -->
    <group ns="$(arg second_robot)">        
        
        <!-- Bringup -->
        <include file="$(find turtlebot3_bringup)/launch/turtlebot3_remote.launch">
            <arg name="multi_robot_name" value="$(arg second_robot)"/>
        </include>

        <!-- Map server -->
        <node name="map_server" pkg="map_server" type="map_server" args="$(find robot_navigation)/maps/office.yaml" />

        <!-- AMCL -->
        <include file="$(find robot_navigation)/launch/amcl_custom.launch">
            <arg name="initial_pose_x"  value="6.5"/>
            <arg name="initial_pose_y"  value="-0.9"/>
            <arg name="initial_pose_a"  value="1.57"/>
            <arg name="ROS_NAMESPACE" value="$(arg second_robot)" />
        </include>

        <!-- Move Base -->
        <include file="$(find robot_navigation)/launch/move_base_custom.launch">
            <arg name="cmd_vel_topic" value="cmd_vel"/>
            <arg name="ROS_NAMESPACE" value="$(arg second_robot)" />
        </include>

    </group>

    <!-- third robot -->
    <group ns="$(arg third_robot)">        
        
        <!-- Bringup -->
        <include file="$(find turtlebot3_bringup)/launch/turtlebot3_remote.launch">
            <arg name="multi_robot_name" value="$(arg third_robot)"/>
        </include>

        <!-- Map server -->
        <node name="map_server" pkg="map_server" type="map_server" args="$(find robot_navigation)/maps/office.yaml" />

        <!-- AMCL -->
        <include file="$(find robot_navigation)/launch/amcl_custom.launch">
            <arg name="initial_pose_x"  value="0"/>
            <arg name="initial_pose_y"  value="3.1"/>
            <arg name="initial_pose_a"  value="0.0"/>
            <arg name="ROS_NAMESPACE" value="$(arg third_robot)" />
        </include>

        <!-- Move Base -->
        <include file="$(find robot_navigation)/launch/move_base_custom.launch">
            <arg name="cmd_vel_topic" value="cmd_vel"/>
            <arg name="ROS_NAMESPACE" value="$(arg third_robot)" />
        </include>

    </group>

    



</launch>